File: CS366/Projects/Proj3/TileCoder_old.py
import math

class TileCoder(object):
    def __init__(self, x1_bounds, x2_bounds, numTilings=4, length=9, numDims=2):
        self.numTilings = numTilings
        self.numDims = numDims
        self.length = length
        self.numTiles = length * length
        self.total_tiles = self.numTiles * self.numTilings
        self.offset = -1.0 / numTilings
        self.max_x1 = x1_bounds[1]
        self.max_x2 = x2_bounds[1]
        self.reset()
        #self.x1 = lengths[0]
        #self.x2 = lengths[1]
    
    def reset(self):
        self.indices = [0] * self.numTilings
    
    def code(self, x_values):
        self.reset()        
        base = 0
        standX1 = (x_values[0] / self.max_x1) * (self.length - 1)
        standX2 = (x_values[1] / self.max_x2) * (self.length - 1)
        
        for i in range(self.numTilings):
            tileNumX1 = int(math.floor(standX1 - base))
            tileNumX2 = int(math.floor(standX2 - base) * self.length)
            tileNum = tileNumX1 + tileNumX2
            self.indices[i] = tileNum + (i * self.numTiles)
            base += self.offset


if __name__ == '__main__':
    bounds = (0, 6)
    numTilings = 8
    length = 11
    coder = TileCoder(bounds, bounds, numTilings, length)
    tests = [(0.1, 0.1), (4.0, 2.0), (5.99, 5.99), (4.0, 2.1)]
    for test in tests:
        coder.code(test)
        print coder.indices

File: CS366/Projects/Proj3/Tilecoder_old.java
public class Tilecoder {
	final static int numOfTilings = 8;
	final static int length = 11;
	final static int numTiles = length * length;
	final static double offset = -1.0 / numOfTilings;
	final static int max_x = 6;

	public static void tilecode(double x1, double x2, int[] tilecodeIndices) {
		double base = 0.0;
		int tileNum; int tileNumX1; int tileNumX2;
		double standardX1 = (x1 / max_x) * (length - 1);
		double standardX2 = (x2 / max_x) * (length - 1);
		
		// compute the tile numbers
		for (int i = 0; i < numOfTilings; i++) {
			
			// find coordinates of activated tile in the tiling space
			tileNumX1 = (int) Math.floor(standardX1 - base);
			tileNumX2 = ((int) Math.floor(standardX2 - base)) * length;
			tileNum = tileNumX1 + tileNumX2;
			tilecodeIndices[i] = tileNum + (i * numTiles);
			// compute offset for next tiling
			base += offset;
		}
	}

	public static void main(String[] args) {
		int[] tilecodeIndices = new int[numOfTilings];

		printTileCoderIndices(0.1, 0.1, tilecodeIndices);
		printTileCoderIndices(4.0, 2.0, tilecodeIndices);
		printTileCoderIndices(5.99, 5.99, tilecodeIndices);
		printTileCoderIndices(4.0, 2.1, tilecodeIndices);
	}

	private static void printTileCoderIndices(double x1, double x2, int[] tilecodeIndices) {
		tilecode(x1, x2, tilecodeIndices);
		System.out.print("Tile indices for input (" + x1 + ", " + x2 + ") are: ");
		for (int i = 0; i < numOfTilings; i++)
			System.out.print(tilecodeIndices[i] + " ");
		System.out.println();
	}
}

